src/std/crypto/libcrypto.ss

```diff
@@ -372,7 +372,7 @@ END-C
 (c-declare #<<END-C
 static HMAC_CTX *ffi_create_HMAC_CTX ()
 {
-#if OPENSSL_VERSION_NUMBER < 0x10100000L ||  defined (LIBRESSL_VERSION_NUMBER)
+#if OPENSSL_VERSION_NUMBER < 0x10100000L
```

Comment:
can we keep this guard around for older versions of LibreSSL?

Reply:
I'm not an expert on using libressl but it seems to be incorrect check in any case:
"if using old OpenSSL or any LibreSSL version -> use old API"

Which versions should use the old API?

Reply:
well it used to not compile with early libressl versions that didnt have the new api.
I am fine removing these guards, just concerned about older openbsd installations.

Reply:
ok, lets merge and if it causes problems for people we can add a more specific (versioned) guard.